core/backend/src/test/standalone/IModelTransformer.test.ts

```diff
@@ -891,4 +891,64 @@ describe("IModelTransformer", () => {
     IModelTransformerUtils.dumpIModelInfo(mergedDb);
     mergedDb.close();
   });
+
+  it("processSchemas should handle out-of-order exported schemas", async () => {
+    const testSchema1Path = IModelTestUtils.prepareOutputFile("IModelTransformer", "TestSchema1.ecschema.xml");
+    IModelJsFs.writeFileSync(testSchema1Path, `<?xml version="1.0" encoding="UTF-8"?>
+      <ECSchema schemaName="TestSchema1" alias="ts1" version="01.00" xmlns="http://www.bentley.com/schemas/Bentley.ECXML.3.1">
+          <ECSchemaReference name="BisCore" version="01.00" alias="bis"/>
+          <ECEntityClass typeName="TestElement1">
+            <BaseClass>bis:PhysicalElement</BaseClass>
+            <ECProperty propertyName="MyProp1" typeName="string"/>
+          </ECEntityClass>
+      </ECSchema>`
+    );
+
+    const testSchema2Path = IModelTestUtils.prepareOutputFile("IModelTransformer", "TestSchema2.ecschema.xml");
+    IModelJsFs.writeFileSync(testSchema2Path, `<?xml version="1.0" encoding="UTF-8"?>
+      <ECSchema schemaName="TestSchema2" alias="ts2" version="01.00.00" xmlns="http://www.bentley.com/schemas/Bentley.ECXML.3.2">
+          <ECSchemaReference name="BisCore" version="01.00.00" alias="bis"/>
+          <ECSchemaReference name="TestSchema1" version="01.00.00" alias="ts1"/>
+          <ECEntityClass typeName="TestElement2">
+            <BaseClass>ts1:TestElement1</BaseClass>
+            <ECProperty propertyName="MyProp2" typeName="string"/>
+          </ECEntityClass>
+      </ECSchema>`
+    );
+
+    const sourceDbPath = IModelTestUtils.prepareOutputFile("IModelTransformer", "OrderTestSource.bim");
+    const sourceDb = SnapshotDb.createEmpty(sourceDbPath, { rootSubject: { name: "Order Test" } });
+
+    const requestContext = new BackendRequestContext();
+    await sourceDb.importSchemas(requestContext, [testSchema1Path, testSchema2Path]);
+    sourceDb.saveChanges();
+
+    class OrderedExporter extends IModelExporter {
+      public async exportSchemas() {
+        const schemaLoader = new IModelSchemaLoader(this.sourceDb);
+        const schema1 = schemaLoader.getSchema("TestSchema1");
+        const schema2 = schemaLoader.getSchema("TestSchema2");
+        // by importing schema2 (which references schema1) first, we
+        // prove that the import order in processSchemas does not matter
+        await this.handler.callProtected.onExportSchema(schema2);
+        await this.handler.callProtected.onExportSchema(schema1);
+      }
+    }
+
+    const targetDbPath = IModelTestUtils.prepareOutputFile("IModelTransformer", "OrderTestTarget.bim");
+    const targetDb = SnapshotDb.createEmpty(targetDbPath, { rootSubject: { name: "Order Test" } });
+    const transformer = new IModelTransformer(new OrderedExporter(sourceDb), targetDb);
+
+    let error: any;
+    try {
+      await transformer.processSchemas(new BackendRequestContext());
+    } catch (_error) {
+      error = _error;
+    }
+    assert.isUndefined(error);

```

Comment:
processSchemas didn't produce an exception. Should you also confirm it imported the schemas successfully?

Reply:
yeah, adding